### ISSUE TITLE
Return type generated for methods using yield allows for null

### ISSUE THREAD

**Issue by y2keeper** [kalessil/phpinspectionsea#1236]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended), 3.0.11 |
| Language level | PHP 7.3.3                     |

Before going into the details let me first thank you for this awesome plugin, it has brought about a near seismic shift in the quality of my code and I genuinely appreciate it :-)

Current behaviour
--
From the method context menu, if you run "Declare the return type" on

    public function testYield()
    {
        yield from [1,2,3];
    }

you end up with:

    public function testYield(): ?\Generator
    {
        yield from [1,2,3];
    }

Note the `?`.

Expected behaviour
--

Return type shouldn't allow for `null`:

    public function testYield(): \Generator
    {
        yield from [1,2,3];
    }

In other words, unless I am mistaken, `yield` never returns `null`. 

For more context, if needed, please refer to my question on SO: https://stackoverflow.com/questions/55158867/why-does-phpstorm-allow-for-null-return-type-on-function-using-yield/55161030 

Environment details
---
PhpStorm 2018.3.5
Build #PS-183.5912.26, built on March 7, 2019
Licensed to \*/\*
Subscription is active until July 23, 2019
JRE: 1.8.0_152-release-1343-b28 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.14.3

**Comment by kalessil** [comment]
Fixed!
